File: src/main/scala/com/kapilkoju/autopos/party/service/VendorService.scala
package com.kapilkoju.autopos.party.service

import com.kapilkoju.autopos.party.domain.Vendor
import org.springframework.data.domain.{Page, Pageable}

trait VendorService {

    def getVendor(id: Long): Option[Vendor]

    def findVendors(query: String, pageable: Pageable): Page[Vendor]

}

File: src/main/kotlin/com/kapilkoju/autopos/party/service/VendorService.kt
package com.kapilkoju.autopos.party.service

import com.kapilkoju.autopos.party.domain.Vendor
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface VendorService {

    fun getVendor(id: Long): Vendor?

    fun findVendors(query: String, pageable: Pageable): Page<Vendor>

}
